spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncherProperties.java

```diff
@@ -43,4 +71,16 @@ public void setModules(String[] modules) {
 		return modules;
 	}
 
+	@PostConstruct
+	public void deduceModuleArguments() throws Exception {
```

Comment:
I agree, but __you__ said
> One thing about the current modules property is ordering - which becomes very important for direct binding. 

which, on second thought I actually don't understand (relation to direct binding). 

Also, sad news, it seems my example above does not bind properly (again, I think because of the _ issue)